datastore/api/snippets.py

```diff
@@ -0,0 +1,823 @@
+            break
+
+        tasks.append(results)
+    # [END cursor_paging]
+
+    return tasks
+
+
+def property_filter(client):
+    # [START property_filter]
+    query = client.query(kind='Task')
+    query.add_filter('done', '=', False)
+    tasks = list(query.fetch())
+    # [END property_filter]
+
+    return tasks
+
+
+def composite_filter(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START composite_filter]
+    query = client.query(kind='Task')
+    query.add_filter('done', '=', False)
+    query.add_filter('priority', '=', 4)
+    tasks = list(query.fetch())
+    # [END composite_filter]
+
+    return tasks
+
+
+def key_filter(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START key_filter]
+    query = client.query(kind='Task')
+    first_key = client.key('Task', 'first_task')
+    query.add_filter('__key__', '>', first_key)
+    tasks = list(query.fetch())
+    # [END key_filter]
+
+    return tasks
+
+
+def ascending_sort(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START ascending_sort]
+    query = client.query(kind='Task')
+    query.order = ['created']
+    tasks = list(query.fetch())
+    # [END ascending_sort]
+
+    return tasks
+
+
+def descending_sort(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START descending_sort]
+    query = client.query(kind='Task')
+    query.order = ['-created']
+    tasks = list(query.fetch())
+    # [END descending_sort]
+
+    return tasks
+
+
+def multi_sort(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START multi_sort]
+    query = client.query(kind='Task')
+    query.order = [
+        '-priority',
+        'created'
+    ]
+    tasks = list(query.fetch())
+    # [END multi_sort]
+
+    return tasks
+
+
+def keys_only_query(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START keys_only_query]
+    query = client.query()
+    query.keys_only()
+    keys = list([entity.key for entity in query.fetch()])
+    # [END keys_only_query]
+
+    return keys
+
+
+def distinct_query(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START distinct_query]
+    query = client.query(kind='Task')
+    query.group_by = ['type', 'priority']
+    query.order = ['type', 'priority']
+    tasks = list(query.fetch())
+    # [END distinct_query]
+
+    return tasks
+
+
+def distinct_on_query(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START distinct_on_query]
+    query = client.query(kind='Task')
+    query.group_by = ['type']
+    query.order = ['type', 'priority']
+    tasks = list(query.fetch())
+    # [END distinct_on_query]
+
+    return tasks
+
+
+def kindless_query(client):
+    # [START kindless_query]
+    an_hour_ago = str(time.time() - 3600)
```

Comment:
Again, I think we need to be consistent across our snippets. In particular here using a timestamp as your primary key is a known datastore anti-pattern.